eureka-client/src/main/java/com/netflix/discovery/converters/Converters.java

```diff
@@ -366,13 +377,13 @@ public Object unmarshal(HierarchicalStreamReader reader,
                 String nodeName = reader.getNodeName();
 
                 if (ELEM_HOST.equals(nodeName)) {
-                    builder.setHostName(reader.getValue());
+                    builder.setHostName(cache.cachedValueOf(reader.getValue()));
                 } else if (ELEM_APP.equals(nodeName)) {
-                    builder.setAppName(reader.getValue());
+                    builder.setAppName(cache.cachedValueOf(reader.getValue()));
                 } else if (ELEM_IP.equals(nodeName)) {
-                    builder.setIPAddr(reader.getValue());
+                    builder.setIPAddr(cache.cachedValueOf(reader.getValue()));
```

Comment:
Are these the best fields for caching? IP for example would be pretty unique for all instances.